### ISSUE TITLE
test.each no-done-callback false positive in version 24.1.1

### ISSUE THREAD

**Issue by CrOrc** [jest-community/eslint-plugin-jest#709]
Plugin version: 24.1.1
Last working version: 24.1.0
Rule **jest/no-done-callback** has false positives with test.each
```
test.each`
  size   | chunkSize    | sum
  ${50}  | ${undefined} | ${(50 * 49) / 2}
`(
  "sums array indices of size $size with chunkSize=$chunkSize => $sum",
  async ({size, chunkSize, sum}) => {
    await expect(
      reduce(new Array(size).fill(0).map((v, index) => index), (ret, it, index) => ret + index, 0, chunkSize)
    ).resolves.toBe(sum);
  }
);
```
